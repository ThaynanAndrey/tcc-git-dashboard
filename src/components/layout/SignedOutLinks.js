import React from 'react';
import { NavLink } from 'react-router-dom';

/**
 * Stateless Component with links to Navbar when user isn't authenticated.
 * 
 * @author Thaynan Nunes
 */
const SignedOutLinks = () => {
    return (
        <ul className="right hide-on-med-and-down">
            <li><NavLink to='/login'>Login</NavLink></li>
        </ul>
    );
};

export default SignedOutLinks;