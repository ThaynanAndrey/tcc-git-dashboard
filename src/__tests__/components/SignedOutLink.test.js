import React from 'react';
import { shallow, configure } from "enzyme";
import toJson from 'enzyme-to-json';

import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

import SignedOutLinks from '../../components/layout/SignedOutLinks';

describe('Test SignedOutLinks', () => {
  it('renders correctly a component SignedOutLinks with classes right hide-on-med-and-down', () => {
    const tree = shallow(
        <SignedOutLinks />
    );

    expect(toJson(tree)).toMatchSnapshot();
    expect(tree.find(".right.hide-on-med-and-down").exists()).toBeTruthy();
  });
})