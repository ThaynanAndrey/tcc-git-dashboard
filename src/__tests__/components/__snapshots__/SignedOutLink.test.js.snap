// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test SignedOutLinks renders correctly a component SignedOutLinks with classes right hide-on-med-and-down 1`] = `
<ul
  className="right hide-on-med-and-down"
>
  <li>
    <NavLink
      activeClassName="active"
      aria-current="page"
      to="/login"
    >
      Login
    </NavLink>
  </li>
</ul>
`;
